### PR TITLE
trivial: Do not abort with a critical warning for a weird TPM event log

### PR DESCRIPTION
**Pull request by hughsie**
There is one eventlog item with zero size.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2038167

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/tpm/fu-tpm-eventlog-parser.c [hunk 1]
```diff
@@ -275,6 +275,8 @@ fu_tpm_eventlog_parser_new(const guint8 *buf,
 						G_LITTLE_ENDIAN,
 						error))
 			return NULL;
+		if (datasz == 0)
+			continue;
```
Comment (superm1) [inline plugins/tpm/fu-tpm-eventlog-parser.c:279]: this is a firmware problem though isn't it?  Shouldn't we make some sort of warning or alert so people know there is a problem to be fixed?

Comment (hughsie) [reply plugins/tpm/fu-tpm-eventlog-parser.c:279]: > this is a firmware problem though isn't it

I'm not sure; a zero-length record isn't explicitly disallowed, it's just a very weird thing to see. Maybe we add a `g_warning("zero length record for PCR %02u type 0x%02x", pcr, event_type);` or something?

Comment (hughsie) [reply plugins/tpm/fu-tpm-eventlog-parser.c:279]: In this case, the event_type is `EV_SEPARATOR` so maybe the zero-length section is actually fine...

Comment (hughsie) [reply plugins/tpm/fu-tpm-eventlog-parser.c:279]: ** (fwupdtpmevlog:145013): DEBUG: 17:06:14.155: Event Data:     01 00 00 00 14 00 00 00 ee 23 1f 8f 2a 42 57 d3 f4 40 43 b8 43 6d ad 7c b5 5b 49 75 
	** (fwupdtpmevlog:145013): DEBUG: 17:06:14.155: Event Data:     53 74 61 72 74 20 4f 70 74 69 6f 6e 20 52 4f 4d 20 53 63 61 6e 
	** (fwupdtpmevlog:145013): DEBUG: 17:06:14.155: Event Data:     07 00 00 00 18 00 00 00 00 00 00 00 15 a6 15 e4 12 fa 28 ca 94 52 30 c1 9d 72 05 bc 4a f2 bb 61 
	** (fwupdtpmevlog:145013): DEBUG: 17:06:14.155: Event Data:     43 61 6c 6c 69 6e 67 20 49 4e 54 20 31 39 68 
	** (fwupdtpmevlog:145013): WARNING **: 17:06:14.155: zero length record for PCR 00 type 0x04
	** (fwupdtpmevlog:145013): WARNING **: 17:06:14.155: zero length record for PCR 01 type 0x04
	** (fwupdtpmevlog:145013): WARNING **: 17:06:14.155: zero length record for PCR 02 type 0x04
	** (fwupdtpmevlog:145013): WARNING **: 17:06:14.155: zero length record for PCR 03 type 0x04
	** (fwupdtpmevlog:145013): WARNING **: 17:06:14.155: zero length record for PCR 04 type 0x04
	** (fwupdtpmevlog:145013): WARNING **: 17:06:14.155: zero length record for PCR 05 type 0x04
	** (fwupdtpmevlog:145013): WARNING **: 17:06:14.155: zero length record for PCR 06 type 0x04
	** (fwupdtpmevlog:145013): WARNING **: 17:06:14.155: zero length record for PCR 07 type 0x04
	** (fwupdtpmevlog:145013): DEBUG: 17:06:14.155: Event Data:     42 6f 6f 74 69 6e 67 20 42 43 56 20 64 65 76 69 63 65 20 38 30 68 20 28 48 44 44 29 

so maybe ->`g_debug`?

Comment (hughsie) [reply plugins/tpm/fu-tpm-eventlog-parser.c:279]: Actually; the patch is wrong; it's still a valid entry, just with no item data.

Comment (hughsie) [reply plugins/tpm/fu-tpm-eventlog-parser.c]: I've reworked all the blob stuff. Thanks for the sanity check!!

**Review by superm1 — APPROVED** [review]

**Pull request merged**